packages/web-api-client/src/a11y-service-client.spec.ts

```diff
@@ -6,34 +6,38 @@ import * as request from 'request-promise';
 import { IMock, Mock, Times } from 'typemoq';
 
 import { A11yServiceClient } from './a11y-service-client';
+import { A11yServiceCredential } from './a11y-service-credential';
 
-// tslint:disable: no-null-keyword
+// tslint:disable: no-null-keyword no-unsafe-any no-any
 describe(A11yServiceClient, () => {
-    const baseUrl = 'base-url';
     let testSubject: A11yServiceClient;
-    let requestMock: IMock<typeof request>;
+    const baseUrl = 'base-url';
     const apiVersion = '1.0';
-    const requestDefaults = {
-        forever: true,
-        qs: {
-            'api-version': apiVersion,
-        },
-        headers: {
-            'Content-Type': 'application/json',
-        },
-    };
+    let credMock: IMock<A11yServiceCredential>;
+    let requestStub: any;
+    let getMock: IMock<(url: string) => {}>;
+    let postMock: IMock<(url: string, options?: request.RequestPromiseOptions) => {}>;
 
     beforeEach(() => {
-        requestMock = Mock.ofType<typeof request>(null);
-        requestMock
-            .setup(req => req.defaults(requestDefaults))
-            .returns(() => requestMock.object)
-            .verifiable(Times.once());
-        testSubject = new A11yServiceClient(baseUrl, apiVersion, requestMock.object);
-    });
+        getMock = Mock.ofInstance(() => {
+            return null;
+        });
+        postMock = Mock.ofInstance(() => {
+            return null;
+        });
+        requestStub = {
+            defaults: (options: request.RequestPromiseOptions) => requestStub,
+            get: getMock.object,
+            post: postMock.object,
+        };
+        credMock = Mock.ofType<A11yServiceCredential>(null);
 
-    afterEach(() => {
-        requestMock.verifyAll();
+        credMock
+            .setup(cm => cm.signRequest(requestStub))
+            .returns(async () => Promise.resolve(requestStub))
+            .verifiable();
```

Comment:
move this signRequest call to helper method & then call it in every test. (setupVerifiableSignRequestCall). this way it is clear that this call is expected in each test.